br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -819,6 +736,20 @@ func (em *ErrorManager) ReplaceConflictKeys(
 			return errors.Trace(err)
 		}
 
+		// delete the additionally inserted rows of nonclustered PK
```

Comment:
can we skip inserting them instead?

Reply:
Theoretically we can, but if we do not insert them, instead we store them using slices or other structures

https://github.com/pingcap/tidb/blob/2d57455a4f9786a11937a8461ebabc5d2cb35add/br/pkg/lightning/errormanager/errormanager.go#L706-L722

Here we need to conduct this same for loop for the stored rows again, it would slow down the performance.